### PR TITLE
DOCS: How to build FreeCAD forked branches

### PR DESCRIPTION
**Pull request by luzpaz**
ref: https://github.com/FreeCAD/FreeCAD-snap/pull/44

Preview: https://github.com/luzpaz/FreeCAD-snap/blob/build-clones/docs/index.md#build-snaps-from-freecad-clones

### CODE REVIEW

**Comment by luzpaz** [comment]
@ppd when you get a chance, would you review ?

**Comment by ppd** [comment]
Hm. The term 'clones' confuses me. Essentially, it just builds a PR and publishes it as a snap branch. It does not need to be based on master at all, and major changes could be made to snapcraft.yaml, too.

Maybe something more like "PRs in this repo can be published as snap branches (link...) to facilitate testing. This can be used to develop the snap itself, but also as a means of building experimental branches or forks of FreeCAD." Then your example.

**Comment by yorikvanhavre** [comment]
Shouldn't this info go to FreeCAD master? In case someone is unaware of this repo, takes the FreeCAD master code and wishes to build a snap package...

**Comment by ppd** [comment]
@yorikvanhavre in that case, it would be better to ship the snapcraft recipe in the main repo.

**Comment by yorikvanhavre** [comment]
Maybe, or, in the docs in master, indicate this repo

**Comment by luzpaz** [comment]
@ppd feel free to rename it. The issue i was having is delineating this feature and the preexisting https://github.com/luzpaz/FreeCAD-snap/blob/build-clones/docs/index.md#publishing-to-branch without being redundant.

**Comment by ppd** [comment]
We should probably remove this publish to branch feature entirely. There is no reason for it to exist anymore; it's much less powerful and a hassle to execute.

**Comment by luzpaz** [comment]
> We should probably remove this publish to branch feature entirely.  

@ppd I can remove it in a separate PR in order not to pollute this one.

File: docs/index.md [hunk 1]
```diff
@@ -77,4 +77,63 @@ $ sudo snap set system experimental.parallel-instances=true
...
 $ sudo snap install freecad_edge --channel=edge
 # run FreeCAD from this parallel install
-$ freecad_edge
\ No newline at end of file
+$ freecad_edge
+```
+
+## Build snaps from FreeCAD clones
+
+The ability build snaps from clones of the FreeCAD master linked to a certain branch is possible. This option is also open to outside users that can simply make a PR to trigger and re-trigger a build.
```
Comment (ppd) [inline docs/index.md:85]:
```suggestion
It is possible to build snaps from any branch of fork of the FreeCAD repository. This option is also open to outside users that can simply make a PR to trigger and re-trigger a build.
```

File: docs/index.md [hunk 2]
```diff
@@ -77,4 +77,63 @@ $ sudo snap set system experimental.parallel-instances=true
 # Install freecad snap from channel edge as 'freecad_edge'
 $ sudo snap install freecad_edge --channel=edge
 # run FreeCAD from this parallel install
-$ freecad_edge
\ No newline at end of file
+$ freecad_edge
+```
+
+## Build snaps from FreeCAD clones
```
Comment (ppd) [inline docs/index.md:83]:
```suggestion
## Build snaps from FreeCAD branches or forks
```

File: docs/index.md [hunk 3]
```diff
@@ -77,4 +77,63 @@ $ sudo snap set system experimental.parallel-instances=true
...
+
+The ability build snaps from clones of the FreeCAD master linked to a certain branch is possible. This option is also open to outside users that can simply make a PR to trigger and re-trigger a build.
+
+1. Clone FreeCAD-snap repo.
```
Comment (ppd) [inline docs/index.md:87]:
```suggestion
1. Fork the FreeCAD-snap repo.
```

File: docs/index.md [hunk 4]
```diff
@@ -77,4 +77,63 @@ $ sudo snap set system experimental.parallel-instances=true
...
+1. Create a new branch to work on.
+1. Open *[snap/snapcraft.yaml](./snap/snapcraft.yaml)* in a text editor
+1. Locate the `freecad` block of the yaml file
+1. Modify the `source:` variable to reflect the specific clone of FreeCAD master to build. Note: must end with `.git` extension.
+1. Add a  `source-branch:` variable to indicate what specific branch of said clone to build.  In the **example below** is a before and after in which we're building a snap of user @WandererFan's `hlrThreadrc1` branch from his FreeCAD clone ([borrowed from actual example](https://github.com/FreeCAD/FreeCAD-snap/pull/44)):
```
Comment (ppd) [inline docs/index.md:92]:
```suggestion
1. Add a  `source-branch:` variable to indicate what specific branch of said fork to build.  In the **example below** is a before and after in which we're building a snap of user @WandererFan's `hlrThreadrc1` branch from his FreeCAD clone ([borrowed from actual example](https://github.com/FreeCAD/FreeCAD-snap/pull/44)):  
```

File: docs/index.md [hunk 5]
```diff
@@ -77,4 +77,63 @@ $ sudo snap set system experimental.parallel-instances=true
...
+      plugin: cmake
+      source: https://github.com/FreeCAD/FreeCAD.git
+    ```
+    ```yaml
+      freecad:
+       plugin: cmake
+       source: https://github.com/WandererFan/FreeCAD.git
+       source-branch: hlrThreadrc1
+1. Save the `snap/snapcraft.yaml` changes to the branch and make Pull Request to FreeCAD-snap.
+1. Assign the 'safe to publish' green tag to the PR.
```
Comment (ppd) [inline docs/index.md:104]:
```suggestion
1. Ask a maintainer to assign the 'safe to publish' green tag to the PR.
```

**Review by ppd — COMMENTED** [review]
Looks good; thanks for that! I suggested a few minor ideas regarding word choice (clone -> fork).

**Comment by ppd** [comment]
> @ppd I can remove it in a separate PR in order not to pollute this one.

@luzpaz Great. Remove the entire workflow & the section in the documentation, please.

**Comment by luzpaz** [comment]
> @luzpaz Great. Remove the entire workflow & the section in the documentation, please.

Thanks! I'll merge this now. And get to work to remove the other section + graphics

**Pull request merged**